### PR TITLE
build(split): add split script

### PR DESCRIPTION
**Pull request by cbourgois**
### Description of the Change

```sh
Usage: split [options] <packageName>
Generate a subtree branch (from the current one) for the specified package
Options:
  -b, --branch [branch]  set a custom branch name
  -p, --push             push the created splitted branch to origin (useless with "remote" option)
  -r, --remote [remote]  remote url (like git@github.com:ovh-ux/manager.git)
```

### Example 

From a `test-split` branch 
```bash
$ git rev-parse --abbrev-ref HEAD
test-split
$ yarn split @ovh-ux/manager-core --push
yarn run v1.12.3
$ node ./scripts/split @ovh-ux/manager-core --push
✔ Search Package infos
✔ Create subtree branch "split-manager-core/test-split" from "@ovh-ux/manager-core" package (packages/manager/modules/core)
✔ Pushing "split-manager-core/test-split" to origin
🎉  And now, you should :

📦  use the splitted branch in your repository :
  yarn add git+https://github.com/ovh-ux/manager.git#split-manager-core/test-split
✨  Done in 7.84s.
````
produces a `split-manager-core/test-split` branch : https://github.com/ovh-ux/manager/tree/split-manager-core/test-split 

872ddb1 — build(split): add split script

/cc @jleveugle @frenautvh @antleblanc

### CODE REVIEW

**Comment by antleblanc** [comment]
Does `$ yarn split @ovh-ux/manager-core --push` produce this following output?
```sh
yarn run v1.12.3
```

According to your example I don't which command a explicitly running.

**Comment by cbourgois** [comment]
On my desk : yes.

`yarn lint:md` produces the following output : 

```bash
yarn run v1.12.3
$ remark -qf .
✨  Done in 1.34s.
```

File: scripts/split.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
+/* eslint-disable no-console */
+const execa = require('execa');
+const gitUrlParse = require('git-url-parse');
+const ora = require('ora');
+const path = require('path');
+const program = require('commander');
+const { find, kebabCase } = require('lodash');
```
Comment (jleveugle) [inline scripts/split.js]:
```suggestion
const find = require('lodash/find');
const kebabCase = require('lodash/kebabCase');
```

File: scripts/split.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+};
+
+const getCurrentBranch = () => execa.shell('git rev-parse --abbrev-ref HEAD')
+  .then(({ stdout: branchName }) => branchName);
+
+const getPackageInfos = searchPackageName => execa.shell('lerna list -la --json')
+  .then(({ stdout: packageList }) => {
+    const searchPackage = find(
+      JSON.parse(packageList),
+      packageInformation => packageInformation.name === searchPackageName,
```
Comment (jleveugle) [inline scripts/split.js]:
```suggestion
      { name : searchPackageName }
```

File: scripts/split.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    }
+    if (searchPackage.private) {
+      return Promise.reject(new Error(`"${searchPackage.name}" is a private package`));
+    }
+    return searchPackage;
+  });
+
+const getSubtreeBranch = (packageName, currentBranch) => {
+  const [scope, name] = packageName.split('/');
+  let leftPart = name;
```
Comment (jleveugle) [inline scripts/split.js]:
```suggestion
  const leftPart = name ? name : scope;
```

File: scripts/split.js (hunk 2 above)
Comment (jleveugle) [reply scripts/split.js]: And `searchPackageName` should be renamed by `name` :)

Keep it simple :)

File: scripts/split.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+  * Split a module :
+    $ split @ovh-ux/manager-core
+
+  * Split a module and push the subtree branch :
+    $ split @ovh-ux/manager-core --push
+
+  * Split a module and specify a branch name :
+    $ split @ovh-ux/manager-core --branch manager-core/split
+
+  * Split a module and push it to another repository :
```
Comment (antleblanc) [inline scripts/split.js]:
```suggestion
  * Split a module and push it to another repository:
```

File: package.json [hunk 5]
```diff
@@ -21,7 +21,8 @@
     "test": "yarn lint && yarn run test:e2e",
     "test:e2e": "lerna run test:e2e --stream --concurrency=1",
     "version": "conventional-changelog -p angular -i CHANGELOG.md -s && git add CHANGELOG.md",
-    "release": "node ./scripts/release"
+    "release": "node ./scripts/release",
+    "split": "node ./scripts/split"
```
[the remark this answers is not in the document]
Comment (antleblanc) [reply package.json]: We probably have to specify a more appropriate script name.

IMO `split` is too generic.

[the remark this answers is not in the document]
Comment (cbourgois) [reply package.json]:
```suggestion
    "split": "scripts/split.js"
```

[the remark this answers is not in the document]
Comment (antleblanc) [reply package.json]: Yes but you can rename your file by removing the `.js` extension.

File: scripts/split.js (hunk 3 above)
Comment (FredericEspiau) [reply scripts/split.js]:
```suggestion
  const leftPart = isString(name) ? name : scope;
```
or
```suggestion
  const leftPart =name || scope;
```
if you don't care about types

File: scripts/split.js [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+const program = require('commander');
+const { find, kebabCase } = require('lodash');
+
+const getBranchUrl = async (branch) => {
+  const { stdout: remoteUrl } = await execa.shell('git config --get remote.origin.url');
+  const gitHttpsUrl = gitUrlParse(remoteUrl).toString('https');
+  return `git+${gitHttpsUrl}#${branch}`;
+};
+
+const getCurrentBranch = () => execa.shell('git rev-parse --abbrev-ref HEAD')
```
Comment (FredericEspiau) [inline scripts/split.js]:
```suggestion
const getCurrentBranch = async () => execa.shell('git rev-parse --abbrev-ref HEAD')
```

This way we know the function returns a `Promise` ;)

**Pull request merged**